webapp/tests/forms/steps/lotse/test_personal_data_steps.py

```diff
@@ -233,3 +232,83 @@ def test_if_no_bundesland_given_then_flash_no_error(self, app):
+    valid_stored_data = {'familienstand': 'married', 'familienstand_date': datetime.date(2000, 1, 31),
+                         'familienstand_married_lived_separated': 'no',
+                         'familienstand_confirm_zusammenveranlagung': True}
+
+    @pytest.fixture()
+    def valid_form_data(self):
+        return {'person_b_idnr': '04452397687', 'person_b_first_name': 'Hermine',
+                'person_b_last_name': 'Granger', 'person_b_dob': ['01', '01', '1985'],
+                'person_b_same_address': 'yes', 'person_b_religion': 'none'}
+
+    def test_if_plz_starts_with_zero_then_succ_validation(self, valid_form_data, new_test_request_context):
+        data = MultiDict({**valid_form_data, ** {'person_b_plz': '01234'}})
+        with new_test_request_context(stored_data=self.valid_stored_data, form_data=data):
+            form = new_person_b_step(form_data=data).render_info.form
+            assert form.validate() is True
+
+    def test_if_gehbeh_and_beh_grad_not_set_then_succ_validation(self, valid_form_data, new_test_request_context):
+        data = MultiDict(valid_form_data)
+        with new_test_request_context(stored_data=self.valid_stored_data, form_data=data):
+            form = new_person_b_step(form_data=data).render_info.form
+            assert form.validate() is True
+
+    def test_if_gehbeh_yes_and_beh_grad_not_set_then_fail_validation(self, valid_form_data, new_test_request_context):
+        data = MultiDict({**valid_form_data, **{'person_b_gehbeh': 'on'}})
+        with new_test_request_context(stored_data=self.valid_stored_data, form_data=data):
+            form = new_person_b_step(form_data=data).render_info.form
+            assert form.validate() is False
+
+    def test_if_gehbeh_yes_and_beh_grad_set_then_succ_validation(self, valid_form_data, new_test_request_context):
+        data = MultiDict({**valid_form_data, **{'person_b_gehbeh': 'on', 'person_b_beh_grad': '30'}})
+        with new_test_request_context(stored_data=self.valid_stored_data, form_data=data):
+            form = new_person_b_step(form_data=data).render_info.form
+            assert form.validate() is True
+
+    def test_if_not_gehbeh_but_beh_grad_set_then_succ_validation(self, valid_form_data, new_test_request_context):
+        data = MultiDict({**valid_form_data, **{'person_b_beh_grad': '30'}})
+        with new_test_request_context(stored_data=self.valid_stored_data, form_data=data):
+            form = new_person_b_step(form_data=data).render_info.form
+            assert form.validate() is True
+
+    def test_if_same_address_yes_then_validation_succ_without_address(self, valid_form_data, new_test_request_context):
+        data = MultiDict({**valid_form_data, **{'person_b_same_address': 'yes'}})
+        with new_test_request_context(stored_data=self.valid_stored_data, form_data=data):
+            form = new_person_b_step(form_data=data).render_info.form
+            assert form.validate() is True
```

Comment:
Do we need to test that the address data is deleted?

Reply:
You mean additionally to the tests in `webapp/tests/model/test_form_data.py`? Why would we need to do that?

Reply:
You are right. We do not need that because we test that already for the SteuerlotseStep in general